test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataSimplifiedOptionsTests.cs

```diff
@@ -0,0 +1,160 @@
+        }
+
+        [Theory]
+        [InlineData(/*SimplifiedOptionsVersion*/ ODataVersion.V4)]
+        [InlineData(/*SimplifiedOptionsVersion*/ ODataVersion.V401)]
+        public void SetV401ODataPrefixWritingTrue(ODataVersion version)
+        {
+            var options = new ODataSimplifiedOptions(version);
+            options.SetOmitODataPrefix(true, ODataVersion.V401);
+            options.GetOmitODataPrefix().Should().Be(version != ODataVersion.V4);
+            options.GetOmitODataPrefix(ODataVersion.V4).Should().BeFalse();
+            options.GetOmitODataPrefix(ODataVersion.V401).Should().BeTrue();
+            options.EnableWritingODataAnnotationWithoutPrefix.Should().Be(version != ODataVersion.V4);
+        }
+
+        [Theory]
+        [InlineData(/*SimplifiedOptionsVersion*/ ODataVersion.V4)]
+        [InlineData(/*SimplifiedOptionsVersion*/ ODataVersion.V401)]
+        public void SetV4ODataPrefixWritingFalse(ODataVersion version)
+        {
+            var options = new ODataSimplifiedOptions(version);
+            options.SetOmitODataPrefix(false, ODataVersion.V4);
+            options.GetOmitODataPrefix().Should().Be(version != ODataVersion.V4);
+            options.GetOmitODataPrefix(ODataVersion.V4).Should().BeFalse();
+            options.GetOmitODataPrefix(ODataVersion.V401).Should().BeTrue();
+            options.EnableWritingODataAnnotationWithoutPrefix.Should().Be(version != ODataVersion.V4);
+        }
+
+        [Theory]
+        [InlineData(/*SimplifiedOptionsVersion*/ ODataVersion.V4)]
+        [InlineData(/*SimplifiedOptionsVersion*/ ODataVersion.V401)]
+
+        public void SetV401ODataPrefixWritingFalse(ODataVersion version)
+        {
+            var options = new ODataSimplifiedOptions(version);
+            options.SetOmitODataPrefix(false, ODataVersion.V401);
+            options.GetOmitODataPrefix().Should().Be(version != ODataVersion.V4);
+            options.GetOmitODataPrefix(ODataVersion.V4).Should().BeFalse();
+            options.GetOmitODataPrefix(ODataVersion.V401).Should().BeFalse();
+            options.EnableWritingODataAnnotationWithoutPrefix.Should().Be(version != ODataVersion.V4);
+        }
+
+        [Theory]
+        [InlineData(/*SimplifiedOptionsVersion*/ ODataVersion.V4)]
+        [InlineData(/*SimplifiedOptionsVersion*/ ODataVersion.V401)]
+
+        public void SetEnableWritingODataAnnotationWithoutPrefixTrue(ODataVersion version)
+        {
+            var options = new ODataSimplifiedOptions(version);
+            options.EnableWritingODataAnnotationWithoutPrefix = true;
+            options.GetOmitODataPrefix().Should().BeTrue();
+            options.GetOmitODataPrefix(ODataVersion.V4).Should().BeTrue();
+            options.GetOmitODataPrefix(ODataVersion.V401).Should().BeTrue();
+            options.EnableWritingODataAnnotationWithoutPrefix.Should().BeTrue();
+        }
+
+        [Fact]
+        public void SetEnableWritingODataAnnotationWithoutPrefixFalse()

```

Comment:
>SetEnableWritingODataAnnotationWithoutPrefixFalse() [](start = 20, length = 51)

why does not it have the InlineData input? By design?

Reply:
This test was intended to test the default instructor; the other tests passed in the version for the constructor. I did add a version of Set...True() that worked with the default constructor, although it was kind of already covered in the defaults test, since true is a default.  Still, good to verify.

---
In reply to: [279948328](https://github.com/OData/odata.net/pull/1454#discussion_r279948328) [](ancestors = 279948328)